spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -82,14 +78,18 @@ private void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer consum
 		}
 
 		try {
-			Message<?> internalMessage =
-					this.messageConverter == null
-							? MessageBuilder.withPayload(pubsubMessage.getData().toByteArray())
-							.copyHeaders(messageHeaders)
-							.build()
-							: this.messageConverter.toMessage(
-									pubsubMessage.getData().toStringUtf8(),
-							new MessageHeaders(messageHeaders));
+			Message<?> internalMessage = null;
+
+			if (this.messageConverter != null) {
+				this.messageConverter.toMessage(
```

Comment:
Should you set `internalMessage` to this?

Reply:
good catch!